crates/ide-completion/src/context.rs

```diff
@@ -796,27 +796,20 @@ impl<'a> CompletionContext<'a> {
             .exclude_traits
             .iter()
             .filter_map(|path| {
-                scope
-                    .resolve_mod_path(&ModPath::from_segments(
-                        hir::PathKind::Plain,
-                        path.split("::").map(Symbol::intern).map(Name::new_symbol_root),
-                    ))
-                    .find_map(|it| match it {
+                hir::resolve_absolute_path(db, path.split("::").map(Symbol::intern)).find_map(
+                    |it| match it {
                         hir::ItemInNs::Types(ModuleDef::Trait(t)) => Some(t),
                         _ => None,
-                    })
+                    },
+                )
```

Comment:
Why this change?

Reply:
Ah I see, that is probably regarding

> Also fix a bug with existing settings for that where the paths wouldn't resolve correctly. @Veykril it seems you just blessed the tests incorrectly in https://github.com/rust-lang/rust-analyzer/pull/18179?

No that was intentional, we need to resolve the paths from the given scope. Though I suppose that `hir::PathKind::Plain` really should've been `hir::PathKind::Absolute`?

Reply:
The test blessing was probably not intentional, as it rendered those tests meaningless.

About

> This method does not make sense, a crate name only exists as a name of a dependency edge. That is we can only resolve paths from within the context of some crate.

Generally I agree, but this is a global setting, so it doesn't depend at the current crate. At the extreme what could happen is that in different crates it will resolve to different things. More problematic, this prevents marking an item from the crate itself, as seen by the tests (without hacks). Also, if some crate reexports another crate, you will have to either write both or let the setting be ignored in some places.

Reply:
Fair enough